From 7122ff1cd6026decef6650bba4f1feaaffa28637 Mon Sep 17 00:00:00 2001
From: Michael Schutte <michi@uiae.at>
Date: Wed, 13 Oct 2010 17:07:26 +0200
Subject: [PATCH 6/6] Fix "compose as usual" for Unicode diacritics

struct ccc in compose_as_usua[() uses three char members of unspecified
signedness; in case the compiler decides that they should be signed,
negative integers will end up in the accent table for c3 > 127.

Avoid the problem by explicitly using unsigned chars.

Signed-off-by: Michael Schutte <michi@uiae.at>
Signed-off-by: Alexey Gladkov <legion@altlinux.org>
---
 src/loadkeys.y |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/loadkeys.y b/src/loadkeys.y
index c8bb7cc..cc7e2ab 100644
--- a/src/loadkeys.y
+++ b/src/loadkeys.y
@@ -1181,7 +1181,7 @@ compose_as_usual(char *charset) {
 		exit(1);
 	} else {
 		struct ccc {
-			char c1, c2, c3;
+			unsigned char c1, c2, c3;
 		} def_latin1_composes[68] = {
 			{ '`', 'A', 0300 }, { '`', 'a', 0340 },
 			{ '\'', 'A', 0301 }, { '\'', 'a', 0341 },
-- 
1.7.3.4

